### PR TITLE
Allow auth node agent to specify TTL

### PR DESCRIPTION
**Pull request by myidpt**
This PR solves: https://github.com/istio/istio/issues/2233
When requesting workload certificate, the node agent should be able to specify TTL with the CSR. The Istio CA has two flags: default workload TTL and max workload TTL. The default workload TTL is applied to the workloads deployed in K8s. The max workload TTL is the cap of the TTL that can be requested by node agents.

### CODE REVIEW

File: security/pkg/pki/ca/ca.go [hunk 1]
```diff
@@ -184,14 +187,21 @@ func (ca *IstioCA) Sign(csrPEM []byte) ([]byte, error) {
 	return chain, nil
 }
 
-func (ca *IstioCA) generateCertificateTemplate(request *x509.CertificateRequest) *x509.Certificate {
+func (ca *IstioCA) generateCertificateTemplate(request *x509.CertificateRequest, requestedTTL time.Duration) *x509.Certificate {
 	exts := append(request.Extensions, request.ExtraExtensions...)
+
+	certTTL := requestedTTL
+	// If the requested TTL is greater than maxCertTTL, apply maxCertTTL as the TTL.
+	if requestedTTL.Seconds() > ca.maxCertTTL.Seconds() {
```
Comment (lookuptable) [inline security/pkg/pki/ca/ca.go]: I'm wondering whether you should reject the request instead of resetting it to max?

Comment (myidpt) [reply security/pkg/pki/ca/ca.go]: That makes sense. I can make it return an error.

**Pull request merged**